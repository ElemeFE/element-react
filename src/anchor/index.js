import AnchorSmooth from './AnchorSmooth';
import GoTop from './GoTop';

export {AnchorSmooth,GoTop};