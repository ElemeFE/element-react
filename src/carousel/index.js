import Carousel from './Carousel';
import CarouselItem from './CarouselItem';

Carousel.Item = CarouselItem;

export default Carousel;
