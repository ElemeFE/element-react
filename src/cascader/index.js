import Cascader from './Cascader';

export default Cascader;
