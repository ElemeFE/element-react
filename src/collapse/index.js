import Collapse from './Collapse';
import CollapseItem from './CollapseItem';

Collapse.Item = CollapseItem;

export default Collapse;
