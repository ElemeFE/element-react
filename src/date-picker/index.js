/*
revision version: 
  48996c6658300b7f9f01b9b4011a531f17e30b75
*/
export { default as TimeSelect } from './TimeSelect'
export { default as TimePicker } from './TimePicker'
export { default as TimeRangePicker } from './TimeRangePicker'
export { default as DatePicker } from './DatePicker'
export { default as DateRangePicker } from './DateRangePicker'