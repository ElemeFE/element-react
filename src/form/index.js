import Form from './Form';
import FormItem from './FormItem';

Form.Item = FormItem;

export default Form;
