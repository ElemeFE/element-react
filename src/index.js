export { default as i18n } from './locale';

export { default as Alert } from './alert';
export { default as Button } from './button';
export { default as Card } from './card';
export { default as Layout } from './layout';
export { default as Loading } from './loading';
export { default as Message } from './message';
export { default as MessageBox } from './message-box';
export { default as Notification } from './notification';
export { default as Radio } from './radio';
export { default as Dialog } from './dialog';
export { default as Rate } from './rate';
export { default as Progress } from './progress';
export { default as Badge } from './badge';
export { default as Tabs } from './tabs';
export { default as Tree } from './tree';
export { default as Input } from './input';
export { default as Icon } from './icon';
export { default as Menu } from './menu';
export { default as Steps } from './steps';
export { default as Breadcrumb } from './breadcrumb';
export { default as Tooltip } from './tooltip';
export { default as InputNumber } from './input-number';
export { default as Checkbox } from './checkbox';
export { default as Slider } from './slider';
export { default as Table } from './table'
export { default as Switch } from './switch';
export { default as Form } from './form';
export { default as Upload } from './upload';
export { default as Tag } from './tag';
export { default as Select } from './select';
export { default as Dropdown } from './dropdown';
export { default as Popover } from './popover';
export { default as Pagination } from './pagination';
export { default as AutoComplete } from './auto-complete';
export { TimeSelect, TimePicker, TimeRangePicker, DatePicker, DateRangePicker } from './date-picker';
export { default as Carousel } from './carousel';
export { default as Collapse } from './collapse';
export { default as ColorPicker } from './color-picker';
export { default as Cascader } from './cascader';
export { default as Transfer } from './transfer';
