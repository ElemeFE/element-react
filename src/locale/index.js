import format from './format';
import defaultLang from './lang/en';

let _lang = defaultLang

function use(lang) {
  _lang = lang;
}

function t(path, options) {
  const array = path.split('.');
    let current = _lang;

    for (var i = 0, j = array.length; i < j; i++) {
      var property = array[i];
      var value = current[property];
      if (i === j - 1) return format(value, options);
      if (!value) return '';
      current = value;
    }
    return '';
}

export default {
  use,
  t
}
