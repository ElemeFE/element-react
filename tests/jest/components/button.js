import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from '../../../src';

test('Basic usage', () => {
  const component = renderer.create(
    <div>
      <Button>Default Button</Button>
      <Button type="primary">Primary Button</Button>
      <Button type="text">Text Button</Button>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Disabled Button', () => {
  const component = renderer.create(
    <div>
      <Button plain={true} disabled={true}>Default Button</Button>
      <Button type="primary" disabled={true}>Primary Button</Button>
      <Button type="text" disabled={true}>Text Button</Button>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Color Indication', () => {
  const component = renderer.create(
    <div className="intro-block">
      <div className="block">
        <span className="demonstration">Default</span>
        <span className="wrapper">
          <Button type="success">Success</Button>
          <Button type="warning">Warning</Button>
          <Button type="danger">Danger</Button>
          <Button type="info">Info</Button>
        </span>
      </div>
      <div className="block">
        <span className="demonstration">Hover to display color</span>
        <span className="wrapper">
          <Button plain={true} type="success">Success</Button>
          <Button plain={true} type="warning">Warning</Button>
          <Button plain={true} type="danger">Danger</Button>
          <Button plain={true} type="info">Info</Button>
        </span>
      </div>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Icon Button', () => {
  const component = renderer.create(
    <div>
      <Button type="primary" icon="edit"></Button>
      <Button type="primary" icon="share"></Button>
      <Button type="primary" icon="delete"></Button>
      <Button type="primary" icon="search">Search</Button>
      <Button type="primary">Upload<i className="el-icon-upload el-icon-right"></i></Button>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Button Group', () => {
  const component = renderer.create(
    <div>
      <Button.Group>
          <Button type="primary" icon="arrow-left">Previous Page</Button>
          <Button type="primary">Next Page<i className="el-icon-arrow-right el-icon-right"></i></Button>
      </Button.Group>
      <Button.Group>
          <Button type="primary" icon="edit"></Button>
          <Button type="primary" icon="share"></Button>
          <Button type="primary" icon="delete"></Button>
      </Button.Group>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Loading Button', () => {
  const component = renderer.create(
    <Button type="primary" loading={true}>Loading</Button>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Sizes', () => {
  const component = renderer.create(
    <div>
      <Button type="primary" size="large">Large Button</Button>
      <Button type="primary">Default Button</Button>
      <Button type="primary" size="small">Small Button</Button>
      <Button type="primary" size="mini">Mini Button</Button>
    </div>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
