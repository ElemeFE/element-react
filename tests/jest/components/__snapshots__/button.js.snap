// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic usage 1`] = `
<div>
  <button
    className="el-button el-button--default"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Default Button
    </span>
  </button>
  <button
    className="el-button el-button--primary"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Primary Button
    </span>
  </button>
  <button
    className="el-button el-button--text"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Text Button
    </span>
  </button>
</div>
`;

exports[`Button Group 1`] = `
<div>
  <div
    className="el-button-group"
    style={Object {}}
  >
    <button
      className="el-button el-button--primary"
      disabled={false}
      onClick={[Function]}
      style={Object {}}
      type="button"
    >
      <i
        className="el-icon-arrow-left"
      />
      <span>
        Previous Page
      </span>
    </button>
    <button
      className="el-button el-button--primary"
      disabled={false}
      onClick={[Function]}
      style={Object {}}
      type="button"
    >
      <span>
        Next Page
        <i
          className="el-icon-arrow-right el-icon-right"
        />
      </span>
    </button>
  </div>
  <div
    className="el-button-group"
    style={Object {}}
  >
    <button
      className="el-button el-button--primary"
      disabled={false}
      onClick={[Function]}
      style={Object {}}
      type="button"
    >
      <i
        className="el-icon-edit"
      />
      <span />
    </button>
    <button
      className="el-button el-button--primary"
      disabled={false}
      onClick={[Function]}
      style={Object {}}
      type="button"
    >
      <i
        className="el-icon-share"
      />
      <span />
    </button>
    <button
      className="el-button el-button--primary"
      disabled={false}
      onClick={[Function]}
      style={Object {}}
      type="button"
    >
      <i
        className="el-icon-delete"
      />
      <span />
    </button>
  </div>
</div>
`;

exports[`Color Indication 1`] = `
<div
  className="intro-block"
>
  <div
    className="block"
  >
    <span
      className="demonstration"
    >
      Default
    </span>
    <span
      className="wrapper"
    >
      <button
        className="el-button el-button--success"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Success
        </span>
      </button>
      <button
        className="el-button el-button--warning"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Warning
        </span>
      </button>
      <button
        className="el-button el-button--danger"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Danger
        </span>
      </button>
      <button
        className="el-button el-button--info"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Info
        </span>
      </button>
    </span>
  </div>
  <div
    className="block"
  >
    <span
      className="demonstration"
    >
      Hover to display color
    </span>
    <span
      className="wrapper"
    >
      <button
        className="el-button el-button--success is-plain"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Success
        </span>
      </button>
      <button
        className="el-button el-button--warning is-plain"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Warning
        </span>
      </button>
      <button
        className="el-button el-button--danger is-plain"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Danger
        </span>
      </button>
      <button
        className="el-button el-button--info is-plain"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Info
        </span>
      </button>
    </span>
  </div>
</div>
`;

exports[`Disabled Button 1`] = `
<div>
  <button
    className="el-button el-button--default is-disabled is-plain"
    disabled={true}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Default Button
    </span>
  </button>
  <button
    className="el-button el-button--primary is-disabled"
    disabled={true}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Primary Button
    </span>
  </button>
  <button
    className="el-button el-button--text is-disabled"
    disabled={true}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Text Button
    </span>
  </button>
</div>
`;

exports[`Icon Button 1`] = `
<div>
  <button
    className="el-button el-button--primary"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <i
      className="el-icon-edit"
    />
    <span />
  </button>
  <button
    className="el-button el-button--primary"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <i
      className="el-icon-share"
    />
    <span />
  </button>
  <button
    className="el-button el-button--primary"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <i
      className="el-icon-delete"
    />
    <span />
  </button>
  <button
    className="el-button el-button--primary"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <i
      className="el-icon-search"
    />
    <span>
      Search
    </span>
  </button>
  <button
    className="el-button el-button--primary"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Upload
      <i
        className="el-icon-upload el-icon-right"
      />
    </span>
  </button>
</div>
`;

exports[`Loading Button 1`] = `
<button
  className="el-button el-button--primary is-loading"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  <i
    className="el-icon-loading"
  />
  <span>
    Loading
  </span>
</button>
`;

exports[`Sizes 1`] = `
<div>
  <button
    className="el-button el-button--primary el-button--large"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Large Button
    </span>
  </button>
  <button
    className="el-button el-button--primary"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Default Button
    </span>
  </button>
  <button
    className="el-button el-button--primary el-button--small"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Small Button
    </span>
  </button>
  <button
    className="el-button el-button--primary el-button--mini"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span>
      Mini Button
    </span>
  </button>
</div>
`;
